### PR TITLE
added (soft) delete and update options + liquibase and spring JPA

### PR DESCRIPTION
**Pull request by annton-shakur**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstoreproject/model/Book.java [hunk 1]
```diff
@@ -28,6 +32,8 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
 
     public Book() {
```
Comment (nshtykh) [inline src/main/java/mate/academy/onlinebookstoreproject/model/Book.java]: Why do you need this constructor if you use Data?

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (nshtykh) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: Add an empty line at the end

**Review by nshtykh — APPROVED** [review]
Great job!

File: src/main/java/mate/academy/onlinebookstoreproject/model/Book.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/mate/academy/onlinebookstoreproject/model/Book.java]: Makes sense, but not critical.

File: src/main/java/mate/academy/onlinebookstoreproject/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -7,15 +7,16 @@
...
+import mate.academy.onlinebookstoreproject.repository.BookRepository;
 import mate.academy.onlinebookstoreproject.service.BookService;
 import org.springframework.stereotype.Service;
 
 @RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private static final String CANNOT_FIND_BOOK_BY_ID_MSG = "Cannot find book by Id: ";
-    private final BookRepositoryImpl bookRepository;
+    private static final String FAILED_TO_UPDATE_BY_ID_MSG = "There is book with the provided ID: ";
```
Comment (vovasalyha) [inline src/main/java/mate/academy/onlinebookstoreproject/service/impl/BookServiceImpl.java]: For me looks like incorrect message. Please check it one more time and think about better one. I would use CANNOT_FIND_BOOK_BY_ID_MSG as the cause is the same, book is not found by ID.

File: src/main/resources/db/changelog/db.changelog-master.yaml (hunk 2 above)
Comment (vovasalyha) [reply src/main/resources/db/changelog/db.changelog-master.yaml]: Not fixed (or some bug on GitHub).

**Review by vovasalyha — APPROVED** [review]
Approving as my comments are minor, but please take a look on them and fix it.

**Pull request merged**